node-tests/addon-tests/addon-test.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+const expect = require('chai').expect;
+const denodeify = require('denodeify');
+const request = denodeify(require('request'));
+const AddonTestApp = require('ember-cli-addon-tests').AddonTestApp;
+const esprima = require('esprima');
+
+describe('ember-cli-addon-tests (slow)', function() {
+  this.timeout(600000);
+
+  let app;
+
+  before(function() {
+    app = new AddonTestApp();
+    let appCreateOptions = {
+      fixturesPath: 'node-tests/addon-tests/fixtures/'
+    }
+    return app.create('ts1', appCreateOptions)
+      .then(() => {
+        return app.run('yarn');
+      })
+      .then(function() {
+        return app.startServer();
+      });
+  });
+
+  after(function() {
+    return app.stopServer();
+  });
+
+  it('transpiles pre-existing typescript files', function() {
+    return request('http://localhost:49741/assets/ts1.js')
+      .then(response => {
+        expect(response.statusCode).to.equal(200);
+        let parsed = esprima.parseScript(response.body);
+        let bodyStatements = parsed.body
```

Comment:
I wonder if we could get away with something like this to make it less scary to maintain/add more tests like this in the future?

```js
let parsed = esprima.parseScript(response.body);
let expected = esprima.parseScript(`
  export function add(a, b) {
    return a + b;
  }
`);

expect(parsed).to.deep.equal(expected);
```

Reply:
@dfreeman Good idea! I'll add that in.